keycloak4s-auth/core/src/main/scala/com/fullfacing/keycloak4s/auth/core/authorization/Authorization.scala

```diff
@@ -0,0 +1,19 @@
+package com.fullfacing.keycloak4s.auth.core.authorization
+
+import java.util.UUID
+
+import com.fullfacing.keycloak4s.auth.core.models.common.PolicyEnforcement
+
+trait Authorization extends PolicyEnforcement {
+
+  val service: String
+
+  /**
+   * Abstract function to handle authorization of a request.
+   *
+   * @param requestPath The path of the HTTP request.
+   * @param method      The HTTP method of the request.
+   * @param userRoles   The permissions of the user.
+   */
+  def authorizeRequest(requestPath: String, method: String, userRoles: List[String])(implicit cId: UUID): Boolean
```

Comment:
Maybe make this generic, allowing the implementer to specify the Request type, which allows access to the full scope of the request.

Reply:
Updated